### PR TITLE
User/grahamc/221003 fix save and remove events

### PR DESCRIPTION
**Pull request by GrahamMcMynn**
For more information about how to contribute to this repo, visit [this page](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md).

## Description
Events are proxied to child windows if no handlers are present in the current window.  However, the default handlers for the save and remove events should not block the proxying of that event.  This change unblocks those 2 events.

### Main changes in the PR:

1. Proxying of save and remove events

## Validation
Tested on the SharePoint Pages app and ensured that the save event was proxied correctly.

### Validation performed:

1.  Loaded SharePoint Pages app to configure a new tab
2. Added a new page
3. Saved

Ensured that save was successfully completed.

### Unit Tests added:
Added unit tests to ensure that save and remove events are proxied to child window if no handler in current window exists, and ensured that no proxying is done if a handler exists.

### End-to-end tests added:

No.

### Change file added:

> Ensure the change file meets the [formatting requirements](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md#change-log-using-beachball).

Yes

### Related PRs:

https://github.com/OfficeDev/microsoft-teams-library-js/pull/1373

### CODE REVIEW

File: packages/teams-js/src/public/pages.ts [hunk 1]
```diff
@@ -442,6 +442,8 @@ export namespace pages {
       const saveEvent = new SaveEventImpl(result);
```
Comment (TrevorJoelHarris) [inline packages/teams-js/src/public/pages.ts:442]: Please add unit tests that will test this new code. The unit tests are in pages.spec.ts, which uses a file called utils.ts to mock the post messaging between the current window and the child window. We should have tests that verify the new behavior (message sent to child window if there is a child window and no one in the current window has registered a handler)

**Review by TrevorJoelHarris — CHANGES_REQUESTED** [review]
Please add unit tests for this change

File: packages/teams-js/src/public/pages.ts (hunk 1 above)
Comment (GrahamMcMynn) [reply packages/teams-js/src/public/pages.ts:442]: I have added unit tests now.

**Review by TrevorJoelHarris — DISMISSED** [review]

File: change/@microsoft-teams-js-27d23c6b-df82-4548-81df-6efc35e67daa.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Proxy save and remove events to child windows",
```
Comment (AE-MS) [inline change/@microsoft-teams-js-27d23c6b-df82-4548-81df-6efc35e67daa.json:3]: >Proxy save and remove events to child windows [](http://example.com/codeflow?start=14&length=45)

For consistency in the public changefile, please use past tense. Something like this:

Enabled proxying of save and remove events in the `pages.config` capability to child windows

**Review by AE-MS — DISMISSED** [review]
:shipit:

**Review by TrevorJoelHarris — DISMISSED** [review]

**Review by TrevorJoelHarris — APPROVED** [review]

**Review by AE-MS — APPROVED** [review]
:shipit:

**Pull request merged**